app/src/XRayServer.cpp

```diff
@@ -0,0 +1,117 @@
+#include "XRayServer.h"
+
+#include <jsonrpc/net/ServerTransport.h>
+#include <jsonrpc/rpc/Dispatcher.h
+
+#include <string>
+
+using namespace jsonrpc;
+
+namespace xray
+{
+
+struct xray::GetValue
+{
+	GetValue(double& value)
+		: _value(value)
+	{}
+	rpc::Json call(const rpc::Json& data) override
+	{
+		return _value;
+	}
+
+private:
+	double& _value;
+};
+
+struct xray::SetValue
+{
+	SetValue(double& value) : _value(value) {}
+	rpc::Json call(const rpc::Json& data) override
+	{
+		_value = data;
+		return {};
+	}
+	private:
+	double& _value;
+};
+
+struct xray::setTubeVoltage
+{
+	rpc::Json call(const rpc::Json& data) override
+	{
+		//TODO: error handling (check if Voltage is in range)
```

Comment:
i think we can skip error handling for the moment. library will reject the call with an error frame, anyway.
`{"error":{"code":-32600,"message":"Invalid Request"},"id":null,"jsonrpc":"2.0"}`